src/annotator/index.js

```diff
@@ -42,63 +40,26 @@ const sidebarLinkElement = /** @type {HTMLLinkElement} */ (
  * client is initially loaded, is also the only guest frame.
  */
 function init() {
-  // Create an internal global used to share data between same-origin guest and
-  // host frames.
-  window_.__hypothesis = {};
-
   const annotatorConfig = getConfig('annotator');
-  const hostFrame = annotatorConfig.subFrameIdentifier ? window.parent : window;
+  const isHostFrame = annotatorConfig.subFrameIdentifier === null;
```

Comment:
As noted in comments in the `Guest` class, I'm not convinced about pushing the knowledge about how to find the host frame in relation to the current frame down into the `Guest` class.

Previously the "is this the host frame?" and "where is the host frame?" logic was located in the same place, in the `init` function. Now it is split across this function and the `Guest` constructor.